asQ/paradiag.py

```diff
@@ -246,10 +260,14 @@ def apply(self, pc, x, y):
 
         # Copy into xfi, xfr
         with self.xfr.dat.vec_wo as v:
-            v.array[:] = parray.real.reshape((self.NM,))
+            # v.array[:] = parray.real.reshape((self.NM,))
+            v.array[:] = parray.real.reshape((np.size(v.array),))
+            v.array[:] = parray.real.reshape((self.blockV.node_set.size * self.M,))
 
         with self.xfi.dat.vec_wo as v:
-            v.array[:] = parray.imag.reshape((self.NM,))
+            # v.array[:] = parray.imag.reshape((self.NM,))
+            v.array[:] = parray.imag.reshape((np.size(v.array),))
+            v.array[:] = parray.imag.reshape((self.blockV.node_set.size * self.M,))
```

Comment:
Two copies seems excessive. If you don't want to figure out the size you could just write `v.array[:] = parray.real.reshape(-1)`